pkg/generators/zync/que_deployment.go

```diff
@@ -0,0 +1,108 @@
+package zync
+
+import (
+	"fmt"
+
+	"github.com/3scale/saas-operator/pkg/basereconciler"
+	"github.com/3scale/saas-operator/pkg/generators/common_blocks/pod"
+	"github.com/3scale/saas-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Deployment returns a basereconciler.GeneratorFunction funtion that will return a Deployment
+// resource when called
+func (gen *QueGenerator) Deployment() basereconciler.GeneratorFunction {
+
+	return func() client.Object {
+
+		dep := &appsv1.Deployment{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Deployment",
+				APIVersion: appsv1.SchemeGroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      gen.GetComponent(),
+				Namespace: gen.Namespace,
+				Labels:    gen.GetLabels(),
+			},
+			Spec: appsv1.DeploymentSpec{
+				Replicas: gen.QueSpec.Replicas,
+				Selector: gen.Selector(),
+				Strategy: appsv1.DeploymentStrategy{
+					Type: appsv1.RollingUpdateDeploymentStrategyType,
+					RollingUpdate: &appsv1.RollingUpdateDeployment{
+						MaxUnavailable: util.IntStrPtr(intstr.FromInt(0)),
+						MaxSurge:       util.IntStrPtr(intstr.FromInt(1)),
+					},
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: gen.LabelsWithSelector(),
+					},
+					Spec: corev1.PodSpec{
+						ImagePullSecrets: func() []corev1.LocalObjectReference {
+							if gen.Image.PullSecretName != nil {
+								return []corev1.LocalObjectReference{{Name: *gen.Image.PullSecretName}}
+							}
+							return nil
+						}(),
+						Containers: []corev1.Container{
+							{
+								Name:  gen.GetComponent(),
+								Image: fmt.Sprintf("%s:%s", *gen.Image.Name, *gen.Image.Tag),
+								Command: func() (commands []string) {
+									commands = []string{
+										"/usr/bin/bash",
+										"-c",
+										"bundle exec rake 'que[--worker-count 10]",
+									}
+									return
+								}(),
```

Comment:
Better not use a function if not necessary:
```suggestion
								Command: []string{
								        "/usr/bin/bash",
								        "-c",
								        "bundle exec rake 'que[--worker-count 10]",
								},

```

Reply:
Good catch! I added it into a function because at the very beginning I thought about being able to modify the worker count via CR, but then I changed my mind (this has never been needed on production), so going to remove it

Reply:
fixed in https://github.com/3scale/saas-operator/pull/74/commits/15a45426846a21d9a12f348501a293c3229b6ea1